runner/internal/tests/tests_test.go

```diff
@@ -0,0 +1,44 @@
+package tests_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/benchttp/engine/runner/internal/metrics"
+	"github.com/benchttp/engine/runner/internal/tests"
+)
+
+func TestRun(t *testing.T) {
+	t.Run("happy path", func(_ *testing.T) {
+		agg := metricsStub()
+		queries := []tests.Case{
+			{
+				Name:      "minimum response time is 50ms", // succeeding case
+				Source:    metrics.ResponseTimeMax,
+				Predicate: tests.GT,
+				Target:    metrics.Value(50 * time.Millisecond),
+			},
+			{
+				Name:      "maximum response time is 110ms", // failing case
+				Source:    metrics.ResponseTimeMax,
+				Predicate: tests.LT,
+				Target:    metrics.Value(110 * time.Millisecond),
+			},
+		}
+		result := tests.Run(agg, queries)
+		fmt.Printf("%+v\n", result)
+	})
+}
```

Comment:
Why is the test not using `testing.T`? Is it even really running?

Reply:
Oh nice catch! I was using it in early development to check the outputs without having to write the actual test.
I'll make the implementation in this PR.